File: Backend/AstroGame.Shared/Models/Technologies/FinishedTechnologies/PlayerDependentFinishedTechnology.cs
using System;
using AstroGame.Shared.Models.Players;

namespace AstroGame.Shared.Models.Technologies.FinishedTechnologies
{
    public abstract class PlayerDependentFinishedTechnology : FinishedTechnology
    {
        public Guid PlayerId { get; set; }

        public virtual Player Player { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/finished_technologies/player_dependent_finished_technology.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/finished_technologies/finished_technology.dart';

import 'package:json_annotation/json_annotation.dart';

@GuidConverter()
abstract class PlayerDependentFinishedTechnology extends FinishedTechnology {
  @JsonKey()
  late Guid playerId;

  PlayerDependentFinishedTechnology();
}
